modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -420,11 +420,10 @@ public function enqueue(): bool {
 	 */
 	public function button_renderer() {
 		$product = wc_get_product();
+
 		if (
 			! is_checkout() && is_a( $product, \WC_Product::class )
-			&& (
-				$product->is_type( array( 'external', 'grouped' ) )
-				|| ! $product->is_in_stock()
+			&& ( ! apply_filters( 'woocommerce_paypal_payments_product_supports_prb', ! $product->is_type( array( 'external', 'grouped' ) ) && $product->is_in_stock(), $product )
```

Comment:
It would be better if we did `apply_filters( 'woocommerce_paypal_payments_product_supports_prb', $product)` instead. This would allow the plugin adding the filter to have access to more info in the product. Also we should add the apply filter as the last part of the if statement. So we don't call the hook unless the other checks passed.

Reply:
Also I tend to dislike abbreviations as sometimes they usually need more context to understand. So I would suggest to change the hook name to `woocommerce_paypal_payments_product_supports_payment_request_button` even though it's quite long.

Reply:
Not sure I understand the first comment. Do you mean something like:
```
$supports = ! is_checkout() && is_a( $product, \WC_Product::class ) && ! $product->is_type( array( 'external', 'grouped' ) ) && $product->is_in_stock();
$supports = apply_filters( 'woocommerce_paypal_payments_product_supports_payment_request_button', $product ); 

if ( ! $supports ) {
	return;
}
```

Reply:
Oops. I misread the `)` here. So we are actually good to go. Just changing the filter name would be enough.